storage/innobase/buf/buf0lru.cc

```diff
@@ -996,6 +998,7 @@ buf_LRU_block_free_non_file_page(
 		      "not perfect alignment");
 	memset_aligned<2>(block->page.frame + FIL_PAGE_ARCH_LOG_NO_OR_SPACE_ID,
 			  0xfe, 4);
+	block->page.set_os_unused();
```

Comment:
Like I pointed out in my previous comment, I think that all the preceding `static_assert` and `memset_aligned` to "invalidate" `block->page.frame` can be omitted. The `MEM_UNDEFINED()` can be kept, to help MemorySanitizer retain some teeth.